fastlane/Helpers/Match.swift

```diff
@@ -33,6 +33,51 @@ enum Match {
                 force: .userDefined(isForce)
             )
         }
+        updateCodeSigning(type: type, appIdentifier: appIdentifier)
+    }
+    
+    static func updateCodeSigning(type: MatchType, appIdentifier: [String]) {
+        // Update Code signing from automatic to manual
+        switch type {
+        case .development:
+            updateCodeSigningSettings(
+                path: Constant.projectPath,
+                useAutomaticSigning: .userDefined(false),
+                targets: .userDefined([Constant.projectName]),
+                buildConfigurations: .userDefined(["Release Staging"]),
+                codeSignIdentity: .userDefined("iPhone Developer"),
+                profileName: .userDefined("match AdHoc \(Constant.stagingBundleId)")
+            )
+        case .adHoc:
+            if appIdentifier.first == Constant.productionBundleId {
```

Comment:
I think you're putting the wrong order for this if else. The `If` should be for production stuff otherwise it's ad-hoc.